github/orgs_outside_collaborators_test.go

```diff
@@ -48,3 +48,97 @@ func TestOrganizationsService_ListOutsideCollaborators_invalidOrg(t *testing.T)
 }
+
+func TestOrganizationsService_RemoveOutsideCollaborator(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
+	}
+	mux.HandleFunc("/orgs/o/outside_collaborators/u", handler)
+
+	_, err := client.Organizations.RemoveOutsideCollaborator(context.Background(), "o", "u")
+	if err != nil {
+		t.Errorf("Organizations.RemoveOutsideCollaborator returned error: %v", err)
+	}
+}
+
+func TestOrganizationsService_RemoveOutsideCollaborator_NonMember(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
+		w.WriteHeader(http.StatusNotFound)
+	}
+	mux.HandleFunc("/orgs/o/outside_collaborators/u", handler)
+
+	_, err := client.Organizations.RemoveOutsideCollaborator(context.Background(), "o", "u")
+	if err, ok := err.(*ErrorResponse); !ok || err.Response.StatusCode != http.StatusNotFound {
+		t.Errorf("Organizations.RemoveOutsideCollaborator_NonMember returned error: %v", err)
+	}
+}
+
+func TestOrganizationsService_RemoveOutsideCollaborator_Member(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
+		w.WriteHeader(http.StatusUnprocessableEntity)
+	}
+	mux.HandleFunc("/orgs/o/outside_collaborators/u", handler)
+
+	_, err := client.Organizations.RemoveOutsideCollaborator(context.Background(), "o", "u")
+	if err, ok := err.(*ErrorResponse); !ok || err.Response.StatusCode != http.StatusUnprocessableEntity {
+		t.Errorf("Organizations.RemoveOutsideCollaborator_Member returned error: %v", err)
+	}
+}
+
+func TestOrganizationsService_ConvertMemberToOutsideCollaborator(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "PUT")
+	}
+	mux.HandleFunc("/orgs/o/outside_collaborators/u", handler)
+
+	_, err := client.Organizations.ConvertMemberToOutsideCollaborator(context.Background(), "o", "u")
+	if err != nil {
+		t.Errorf("Organizations.ConvertMemberToOutsideCollaborator returned error: %v", err)
+	}
+}
+
+func TestOrganizationsService_ConvertMemberToOutsideCollaborator_NonMember(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "PUT")
+		w.WriteHeader(http.StatusForbidden)
+	}
+	mux.HandleFunc("/orgs/o/outside_collaborators/u", handler)
+
+	_, err := client.Organizations.ConvertMemberToOutsideCollaborator(context.Background(), "o", "u")
+	if err, ok := err.(*ErrorResponse); !ok || err.Response.StatusCode != http.StatusForbidden {
+		t.Errorf("Organizations.ConvertMemberToOutsideCollaborator_NonMember returned error: %v", err)
+	}
+}
+
+func TestOrganizationsService_ConvertMemberToOutsideCollaborator_LastOwner(t *testing.T) {
```

Comment:
AFAICT, `TestOrganizationsService_ConvertMemberToOutsideCollaborator_LastOwner` and `TestOrganizationsService_ConvertMemberToOutsideCollaborator_NonMember` are identical, so I don't see the value in keeping them both.

Can you please delete one and maybe just add a comment that the two cases appear identical?

Reply:
> AFAICT, TestOrganizationsService_ConvertMemberToOutsideCollaborator_LastOwner and TestOrganizationsService_ConvertMemberToOutsideCollaborator_NonMember are identical, so I don't see the value in keeping them both.

I agree with you that the current implementation of the tests is identical in nature. My motivation behind writing two separate tests was to illustrate the fact that both the cases are handled differently by the API. The HTTP response codes are the same (403) but the `message` fields are different, based on the scenario.

Reply:
This was the reason I asked for some guideline in the preceding comment.

Reply:
OK, I think a comment is sufficient to explain the situation. We don't need tests to show every possible response from the server, so please delete the duplicated test. As for how the remaining test is written, it LGTM as-is. An added comment can explain what is going on... it can be quick and short... no need to go into great detail.

Thanks, @kshitij10496!